From b338127530a12e742e85571782ec7cb6bda144c9 Mon Sep 17 00:00:00 2001
From: Kaizhen <kli@mozilla.com>
Date: Sun, 19 Jan 2014 11:13:33 +0800
Subject: [PATCH] Bug-959054-Do-not-launch-the-usage-widget-frame-when start

---
 apps/system/js/cost_control.js |    2 --
 1 file changed, 2 deletions(-)

diff --git a/apps/system/js/cost_control.js b/apps/system/js/cost_control.js
index 9f817be..fd4f8a7 100644
--- a/apps/system/js/cost_control.js
+++ b/apps/system/js/cost_control.js
@@ -113,11 +113,9 @@
       if (enabled !== false) {
         window.addEventListener('ftudone', function ftudone(e) {
           window.removeEventListener('ftudone', ftudone);
-          _ensureWidget();
           widgetFrame.setVisible(false);
         });
       } else {
-        _ensureWidget();
         widgetFrame.setVisible(false);
       }
     });
-- 
1.7.9.5

