From 996d6ad1cdddaf8113eab943e44e5b35b067f156 Mon Sep 17 00:00:00 2001
From: Kai-Zhen Li <kli@mozilla.com>
Date: Tue, 21 Jan 2014 13:27:32 +0800
Subject: [PATCH 1/2] Do not start usage in start.

---
 apps/system/js/cost_control.js |    2 --
 1 file changed, 2 deletions(-)

diff --git a/apps/system/js/cost_control.js b/apps/system/js/cost_control.js
index 9f817be..fd4f8a7 100644
--- a/apps/system/js/cost_control.js
+++ b/apps/system/js/cost_control.js
@@ -113,11 +113,9 @@
       if (enabled !== false) {
         window.addEventListener('ftudone', function ftudone(e) {
           window.removeEventListener('ftudone', ftudone);
-          _ensureWidget();
           widgetFrame.setVisible(false);
         });
       } else {
-        _ensureWidget();
         widgetFrame.setVisible(false);
       }
     });
-- 
1.7.9.5

