From 385d5e5c068dd52fdc8766a0b7c565b88fb722ee Mon Sep 17 00:00:00 2001
From: Kai-Zhen Li <kli@mozilla.com>
Date: Tue, 21 Jan 2014 13:27:54 +0800
Subject: [PATCH 2/2] Install gaia in data.

---
 Android.mk |    4 ++--
 Makefile   |    2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/Android.mk b/Android.mk
index 81f29d9..e0cd1b8 100644
--- a/Android.mk
+++ b/Android.mk
@@ -25,7 +25,7 @@ CLEAN_PROFILE := 0
 # In user (production) builds we put gaia apps in /system/b2g/webapps
 ifneq ($(filter user userdebug, $(TARGET_BUILD_VARIANT)),)
 GAIA_MAKE_FLAGS += PRODUCTION=1
-B2G_SYSTEM_APPS := 1
+#B2G_SYSTEM_APPS := 1
 endif
 
 # Gaia currently needs to specify the default scale value manually or pictures
@@ -35,7 +35,7 @@ GAIA_MAKE_FLAGS += GAIA_DEV_PIXELS_PER_PX=$(GAIA_DEV_PIXELS_PER_PX)
 endif
 
 ifeq ($(B2G_SYSTEM_APPS),1)
-GAIA_MAKE_FLAGS += B2G_SYSTEM_APPS=1
+#GAIA_MAKE_FLAGS += B2G_SYSTEM_APPS=1
 GAIA_APP_INSTALL_PARENT := $(TARGET_OUT)/b2g
 CLEAN_PROFILE := 1
 endif
diff --git a/Makefile b/Makefile
index cac765d..f6e59ce 100644
--- a/Makefile
+++ b/Makefile
@@ -144,7 +144,7 @@ MARIONETTE_RUNNER_HOST?=marionette-b2gdesktop-host
 TEST_MANIFEST?=./shared/test/integration/travis-manifest.json
 MOZPERFOUT?=""
 
-GAIA_INSTALL_PARENT?=/system/b2g
+GAIA_INSTALL_PARENT?=/data/local
 
 ifeq ($(MAKECMDGOALS), demo)
 GAIA_DOMAIN=thisdomaindoesnotexist.org
-- 
1.7.9.5

