From 708a8a225fbe64e0db40cdd30d396a0e5367b3bb Mon Sep 17 00:00:00 2001
From: Kai-Zhen Li <kli@mozilla.com>
Date: Fri, 17 Jan 2014 10:45:06 +0800
Subject: [PATCH] Update zram swapiness to 100.

---
 sp6821a_gonk/init.sp8810.rc |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sp6821a_gonk/init.sp8810.rc b/sp6821a_gonk/init.sp8810.rc
index cf76078..ecf5938 100644
--- a/sp6821a_gonk/init.sp8810.rc
+++ b/sp6821a_gonk/init.sp8810.rc
@@ -74,7 +74,7 @@ on property:ro.debuggable=1
 
 on property:zram.support=true
     setprop zram.disksize 64
-    setprop sys.vm.swappiness 60
+    setprop sys.vm.swappiness 100
     start zram
 
 service zram /system/xbin/zram.sh
-- 
1.7.9.5

