From c1d8fc229755086df8cb8a6d959519f8b386c5f3 Mon Sep 17 00:00:00 2001
From: dannyliang <dliang@mozilla.com>
Date: Thu, 6 Mar 2014 12:01:22 +0800
Subject: [PATCH 2/2] Bug 972130 - tune the memory pressure threshold

---
 sp6821a_gonk/dev-pref.js |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sp6821a_gonk/dev-pref.js b/sp6821a_gonk/dev-pref.js
index 4e052cd..3af6d0c 100644
--- a/sp6821a_gonk/dev-pref.js
+++ b/sp6821a_gonk/dev-pref.js
@@ -19,4 +19,4 @@ pref("hal.processPriorityManager.gonk.BACKGROUND_PERCEIVABLE.KillUnderKB", 6144)
 
 pref("hal.processPriorityManager.gonk.BACKGROUND.KillUnderKB", 20480);
 
-pref("hal.processPriorityManager.gonk.notifyLowMemUnderKB", 10240);
+pref("hal.processPriorityManager.gonk.notifyLowMemUnderKB", 1600);
-- 
1.7.9.5

