diff --git a/sp6821a_gonk/init.rc b/sp6821a_gonk/init.rc
index 65c315a..8c79324 100644
--- a/sp6821a_gonk/init.rc
+++ b/sp6821a_gonk/init.rc
@@ -245,6 +245,7 @@ on boot
 # parameters to match how it is managing things.
     write /proc/sys/vm/overcommit_memory 1
     write /proc/sys/vm/min_free_order_shift 4
+    write /proc/sys/vm/min_free_kbytes 512
     chown root system /sys/module/lowmemorykiller/parameters/adj
     chmod 0664 /sys/module/lowmemorykiller/parameters/adj
     chown root system /sys/module/lowmemorykiller/parameters/minfree
