From 0f0e170ecf4bb32acc7606f21f4d0d062f9f75e3 Mon Sep 17 00:00:00 2001
From: Kai-Zhen Li <kli@mozilla.com>
Date: Tue, 22 Apr 2014 16:45:30 +0800
Subject: [PATCH] Update LMK.

---
 sp6821a_gonk/dev-pref.js |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/sp6821a_gonk/dev-pref.js b/sp6821a_gonk/dev-pref.js
index 32240b1..7f4ed19 100644
--- a/sp6821a_gonk/dev-pref.js
+++ b/sp6821a_gonk/dev-pref.js
@@ -12,3 +12,5 @@ pref("hal.processPriorityManager.gonk.enableOOBPKiller", true);
 pref("geo.provider.use_mls", true);
 pref("geo.wifi.uri", "https://location.services.mozilla.com/v1/geolocate?key=%MOZILLA_API_KEY%");
 pref("geo.cell.scan", true);
+
+pref("hal.processPriorityManager.gonk.BACKGROUND_HOMESCREEN.KillUnderKB", 16384);
-- 
1.7.9.5

