From 7d52476ff1f9c85363f0f8c81da04ee677859b0a Mon Sep 17 00:00:00 2001
From: Kai-Zhen Li <kli@mozilla.com>
Date: Tue, 22 Apr 2014 17:18:14 +0800
Subject: [PATCH] Disable compress fonts.

---
 sp6821a_gonk/prod_sp6821a.mk |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/sp6821a_gonk/prod_sp6821a.mk b/sp6821a_gonk/prod_sp6821a.mk
index b8ef505..be15421 100644
--- a/sp6821a_gonk/prod_sp6821a.mk
+++ b/sp6821a_gonk/prod_sp6821a.mk
@@ -65,10 +65,10 @@ PRODUCT_MODEL := sp6821a
 PRODUCT_BRAND := Spreadtrum
 PRODUCT_MANUFACTURER := Spreadtrum
 
-PRODUCT_LOCALES := zh_CN zh_TW en_US
+MOZ_PRODUCT_LOCALES := bn-BD
 
 # use woff font for ROM size
-MOZ_PRODUCT_COMPRESS_FONTS := true
+MOZ_PRODUCT_COMPRESS_FONTS := false
 
 # Gecko/Gaia need below export value
 # Using Mozilla API key to support MLS
-- 
1.7.9.5

