From 61ec9f0993d79ce3c3b0f5f986c84d2f02e46cb9 Mon Sep 17 00:00:00 2001
From: dannyliang <dliang@mozilla.com>
Date: Fri, 14 Mar 2014 15:16:01 +0800
Subject: [PATCH] Bug 972130 - [tarako] tune the memory pressure threshold

---
 drivers/staging/android/lowmemorykiller.c |    9 +++------
 1 file changed, 3 insertions(+), 6 deletions(-)

diff --git a/drivers/staging/android/lowmemorykiller.c b/drivers/staging/android/lowmemorykiller.c
index b42e31d..efa32ae 100644
--- a/drivers/staging/android/lowmemorykiller.c
+++ b/drivers/staging/android/lowmemorykiller.c
@@ -143,8 +143,8 @@ static int lowmem_shrink(struct shrinker *s, struct shrink_control *sc)
 
 	get_free_ram(&other_free, &other_file);
 
-	if (other_free < lowmem_minfree_notif_trigger &&
-			other_file < lowmem_minfree_notif_trigger) {
+	if (sc->nr_to_scan > 0 &&
+			nr_free_pages() < lowmem_minfree_notif_trigger) {
 		lowmem_notify_killzone_approach();
 	}
 
@@ -243,11 +243,8 @@ static void lowmem_notify_killzone_approach(void)
 static ssize_t lowmem_notify_trigger_active_show(struct kobject *k,
 		struct kobj_attribute *attr, char *buf)
 {
-	int other_free, other_file;
 	lowmem_print(5, "lowmem_shrink notify_trigger_active_show\n");
-	get_free_ram(&other_free, &other_file);
-	if (other_free < lowmem_minfree_notif_trigger &&
-			other_file < lowmem_minfree_notif_trigger)
+	if (nr_free_pages() < lowmem_minfree_notif_trigger)
 		return snprintf(buf, 3, "1\n");
 	else
 		return snprintf(buf, 3, "0\n");
-- 
1.7.9.5

