From 3caec1fff9a0aca4dc25292d913c9aed54c3abc4 Mon Sep 17 00:00:00 2001
From: Kaizhen <kli@mozilla.com>
Date: Mon, 13 Jan 2014 00:11:30 +0800
Subject: [PATCH 10/12] Bug 897745 - Implement the CSP cache using ES6 Map.

---
 content/base/src/contentSecurityPolicy.js |   12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/content/base/src/contentSecurityPolicy.js b/content/base/src/contentSecurityPolicy.js
index 03e7236..401386a 100644
--- a/content/base/src/contentSecurityPolicy.js
+++ b/content/base/src/contentSecurityPolicy.js
@@ -56,7 +56,7 @@ function ContentSecurityPolicy() {
   this._weakDocRequest = { get : function() { return null; } };
   CSPdebug("CSP object initialized, no policies to enforce yet");
 
-  this._cache = { };
+  this._cache = new Map();
 }
 
 /*
@@ -372,7 +372,7 @@ ContentSecurityPolicy.prototype = {
     newpolicy._specCompliant = !!aSpecCompliant;
     newpolicy._isInitialized = true;
     this._policies.push(newpolicy);
-    this._cache = {}; // reset cache since effective policy changes
+    this._cache.clear(); // reset cache since effective policy changes
   },
 
   /**
@@ -385,7 +385,7 @@ ContentSecurityPolicy.prototype = {
       return;
     }
     this._policies.splice(index, 1);
-    this._cache = {}; // reset cache since effective policy changes
+    this._cache.clear(); // reset cache since effective policy changes
   },
 
   /**
@@ -659,8 +659,8 @@ ContentSecurityPolicy.prototype = {
                           aMimeTypeGuess,
                           aOriginalUri) {
     let key = this._createCacheKey(aContentLocation, aContentType);
-    if (key && this._cache[key]) {
-      return this._cache[key];
+    if (key && this._cache.has(key)) {
+      return this._cache.get(key);
     }
 
 #ifndef MOZ_B2G
@@ -764,7 +764,7 @@ ContentSecurityPolicy.prototype = {
 
     // Do not cache the result if this is a nonce-source preload
     if (key && !possiblePreloadNonceConflict) {
-      this._cache[key] = ret;
+      this._cache.set(key, ret);
     }
     return ret;
   },
-- 
1.7.9.5

