From 1572c0917e70d5c7657cb97eb85d8dc00f95a060 Mon Sep 17 00:00:00 2001
From: Kaizhen <kli@mozilla.com>
Date: Sat, 11 Jan 2014 17:02:49 +0800
Subject: [PATCH 02/12] Bug xxxxxx - disable hw composer.

---
 b2g/app/b2g.js |    1 +
 1 file changed, 1 insertion(+)

diff --git a/b2g/app/b2g.js b/b2g/app/b2g.js
index ebcd9a7..3a9b839 100644
--- a/b2g/app/b2g.js
+++ b/b2g/app/b2g.js
@@ -268,6 +268,7 @@ pref("layers.offmainthreadcomposition.async-animations", true);
 pref("layers.async-video.enabled", true);
 pref("layers.async-pan-zoom.enabled", true);
 pref("gfx.content.azure.backends", "cairo");
+pref("layers.composer2d.enabled", false);
 #endif
 
 // Web Notifications
-- 
1.7.9.5

