From d251864b80d94bdce64a5b480b9dacf88235c246 Mon Sep 17 00:00:00 2001
From: vincentliu <vliu@mozilla.com>
Date: Fri, 7 Feb 2014 12:19:08 +0800
Subject: [PATCH] Bug 948828 - Set a hard limit for the number background
 processes

---
 dom/ipc/ProcessPriorityManager.cpp |   20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/dom/ipc/ProcessPriorityManager.cpp b/dom/ipc/ProcessPriorityManager.cpp
index dcd1de8..9b50c4f 100644
--- a/dom/ipc/ProcessPriorityManager.cpp
+++ b/dom/ipc/ProcessPriorityManager.cpp
@@ -318,6 +318,7 @@ public:
 private:
   static StaticAutoPtr<BackgroundProcessLRUPool> sSingleton;
 
+  bool mEnableOOBPKiller;
   int32_t mLRUPoolLevels;
   int32_t mLRUPoolSize;
   int32_t mLRUPoolAvailableIndex;
@@ -1258,6 +1259,17 @@ BackgroundProcessLRUPool::EnsureLRUPool()
   LOG("Making background LRU pool with size(%d)", mLRUPoolSize);
 
   mLRUPool.InsertElementsAt(0, mLRUPoolSize, (ContentParent*)nullptr);
+
+  // We set mEnableOOBPKiller according to our pref.
+  // If this value is true, we will call ContentParent-<KillHard() to kill
+  // background process in LRU order if we hit the background LRU pool's limit.
+  if (!NS_SUCCEEDED(Preferences::GetBool(
+        "hal.processPriorityManager.gonk.enableOOBPKiller",
+        &mEnableOOBPKiller))) {
+    mEnableOOBPKiller = false;
+  }
+  LOG("Out of background process killer is %s",
+      mEnableOOBPKiller ? "enabled" : "disabled");
 }
 
 void
@@ -1325,6 +1337,14 @@ BackgroundProcessLRUPool::UpdateAvailableIndexInLRUPool(
     }
   }
 
+  // Kill background process in LRU order if we run out of background LRU pool
+  // vacancy
+  if (mEnableOOBPKiller && mLRUPoolAvailableIndex == -1) {
+    LOG("Out of background process, killing ChildID(%llu)",
+        mLRUPool[mLRUPoolSize - 1]->ChildID());
+    mLRUPool[mLRUPoolSize - 1]->KillHard();
+  }
+
   // If the LRUPool is already full, mLRUPoolAvailableIndex is still -1 after
   // above loop finished. We should set mLRUPoolAvailableIndex
   // to mLRUPoolSize - 1 in this case. Here uses the mod operator to do it:
-- 
1.7.9.5

