From 79698020a89fdf1263e7740056ecfc49b00ea906 Mon Sep 17 00:00:00 2001
From: dliang <dliang@mozilla.com>
Date: Tue, 11 Feb 2014 10:49:13 +0800
Subject: [PATCH] Correct gralloc setting to fix no video issue

---
 gfx/layers/GrallocImages.h |    8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/gfx/layers/GrallocImages.h b/gfx/layers/GrallocImages.h
index 742d6fb..99800bc 100644
--- a/gfx/layers/GrallocImages.h
+++ b/gfx/layers/GrallocImages.h
@@ -112,11 +112,11 @@ public:
   // From [android 4.0.4]/hardware/msm7k/libgralloc-qsd8k/gralloc_priv.h
   enum {
     /* OEM specific HAL formats */
-    HAL_PIXEL_FORMAT_YCbCr_422_P            = 0x102,
-    HAL_PIXEL_FORMAT_YCbCr_420_P            = 0x103,
-    HAL_PIXEL_FORMAT_YCbCr_420_SP           = 0x109,
+    HAL_PIXEL_FORMAT_YCbCr_422_P            = 0x12,
+    HAL_PIXEL_FORMAT_YCbCr_420_P            = 0x13,
+    HAL_PIXEL_FORMAT_YCbCr_420_SP           = 0x21,
     HAL_PIXEL_FORMAT_YCrCb_420_SP_ADRENO    = 0x10A,
-    HAL_PIXEL_FORMAT_YCbCr_420_SP_TILED     = 0x7FA30C03,
+    HAL_PIXEL_FORMAT_YCbCr_420_SP_TILED     = 0x20,
     HAL_PIXEL_FORMAT_YCbCr_420_SP_VENUS     = 0x7FA30C04,
   };
 
-- 
1.7.9.5

