From 686fb7830ba716ea42e8b2b39f3b06aaf965e4bc Mon Sep 17 00:00:00 2001
From: dannyliang <dliang@mozilla.com>
Date: Thu, 13 Feb 2014 20:53:37 +0800
Subject: [PATCH] disable orientation

---
 widget/gonk/OrientationObserver.cpp |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/widget/gonk/OrientationObserver.cpp b/widget/gonk/OrientationObserver.cpp
index dd70293..6d2ec74 100644
--- a/widget/gonk/OrientationObserver.cpp
+++ b/widget/gonk/OrientationObserver.cpp
@@ -239,11 +239,11 @@ OrientationObserver::Notify(const hal::SensorData& aSensorData)
 void
 OrientationObserver::EnableAutoOrientation()
 {
-  MOZ_ASSERT(NS_IsMainThread() && !mAutoOrientationEnabled);
+/*  MOZ_ASSERT(NS_IsMainThread() && !mAutoOrientationEnabled);
 
   mOrientation->Reset();
   hal::RegisterSensorObserver(hal::SENSOR_ACCELERATION, this);
-  mAutoOrientationEnabled = true;
+  mAutoOrientationEnabled = true;*/
 }
 
 /**
-- 
1.7.9.5

